File: Day12/descending_test.py
import unittest
from descending_order import *


class MyTestCase(unittest.TestCase):
    def test_that_number_can_be_sorted_from_ascending_order_to_descending_ordee(self):
        number = 12345
        actual = descending_order(number)
        expected = '54321'
        self.assertEqual(actual, expected)

    def test_that_number_cannot_be_negative_and_raise_value_error(self):
        self.assertRaises(ValueError, descending_order, -1)





if __name__ == '__main__':
    unittest.main()

File: Day12/DescendingTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class DescendingTest{

@Test
public void testThatDescendingWorksAsExpected(){
	DescendingOrder descending = new DescendingOrder();
	
	int number = 12345;
	String actual = descending.Descending(number);
	String expected = "54321";
	assertEquals(actual,expected);


}

@Test
public void testThatSoretedArrayCanBeSorted(){
	DescendingOrder descending = new DescendingOrder();
	int [] number = {1,2,3,4,5};
	int []actual = descending.sortArray(number);
	int [] expected = {5,4,3,2,1};
	assertArrayEquals(actual, expected);


}


@Test
public void testThatJoinedArrayCanBeJoined(){
	DescendingOrder descending = new DescendingOrder();
	int [] number = {1,2,3,4,5};
	String  actual = descending.joinedNumbers(number);
	String expected = "12345";
	assertEquals(actual,expected);



}


}